src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IAttributeOperation.cs

```diff
@@ -0,0 +1,883 @@
+          OutConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+    Initializer:
+      null
+";
+            VerifyOperationTreeAndDiagnosticsForTest<AttributeSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact]
+        public void AttributeWithTypeOfArgument()
+        {
+            string source = @"
+using System;
+
+[/*<bind>*/My(typeof(MyAttribute))/*</bind>*/]
+class MyAttribute : Attribute
+{
+    public MyAttribute(Type opt = null)
+    {
+    }
+}
+";
+            var expectedDiagnostics = DiagnosticDescription.None;
+            string expectedOperationTree = @"
+IAttributeOperation (OperationKind.Attribute, Type: null) (Syntax: 'My(typeof(MyAttribute))')
+  IObjectCreationOperation (Constructor: MyAttribute..ctor([System.Type opt = null])) (OperationKind.ObjectCreation, Type: MyAttribute, IsImplicit) (Syntax: 'My(typeof(MyAttribute))')
+    Arguments(1):
+        IArgumentOperation (ArgumentKind.Explicit, Matching Parameter: opt) (OperationKind.Argument, Type: null) (Syntax: 'typeof(MyAttribute)')
+          ITypeOfOperation (OperationKind.TypeOf, Type: System.Type) (Syntax: 'typeof(MyAttribute)')
+            TypeOperand: MyAttribute
+          InConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+          OutConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+    Initializer:
+      null
+";
+            VerifyOperationTreeAndDiagnosticsForTest<AttributeSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [Fact]
+        public void InvalidValue()
+        {
+            string source = @"
+using System.Security.Permissions;
+
+[/*<bind>*/A/*</bind>*/]
+class A : CodeAccessSecurityAttribute
+{
+    public A(SecurityAction a = 0) : base(a)
+    {
+    }
+
+}
+";
+            var expectedDiagnostics = new DiagnosticDescription[]
+            {
+                // (4,12): error CS7049: Security attribute 'A' has an invalid SecurityAction value '0'
+                // [/*<bind>*/A/*</bind>*/]
+                Diagnostic(ErrorCode.ERR_SecurityAttributeInvalidAction, "A").WithArguments("A", "0").WithLocation(4, 12),
+                // (5,7): error CS0534: 'A' does not implement inherited abstract member 'SecurityAttribute.CreatePermission()'
+                // class A : CodeAccessSecurityAttribute
+                Diagnostic(ErrorCode.ERR_UnimplementedAbstractMethod, "A").WithArguments("A", "System.Security.Permissions.SecurityAttribute.CreatePermission()").WithLocation(5, 7)
+            };
+            // Can we get this to produce IInvalidOperation?
```

Comment:
Is this still a question?

Reply:
I'll delete the comment. We don't want to produce IInvalidOperation here.